Create HilbertSandTrap.lua

R-Nukem opened the pull request:
Adding today's map so everyone has access.

R-Nukem committed:
Create HilbertSandTrap.lua

Adding today's map so everyone has access.

R-Nukem committed:
Update HilbertSandTrap.lua

R-Nukem committed:
Update HilbertSandTrap.lua

R-Nukem committed:
Update HilbertSandTrap.lua

grilledham commented on map_gen/maps/HilbertSandTrap.lua:
```diff
@@ -0,0 +1,287 @@
+Event.add(
+    defines.events.on_player_built_tile,
+    function(event)
+		local player = (require 'utils.game').get_player_by_index(event.player_index)
```
I wrote the code this way because I had to for the hotfix (no upvaules in removable event function).
`(require 'utils.game')` should be changed to `Game`

grilledham commented on map_gen/maps/HilbertSandTrap.lua:
```diff
@@ -0,0 +1,287 @@
+		if #replace_tiles > 0 then
+			player.surface.set_tiles(replace_tiles, true)
+		end
+		player.surface.set_tiles(replace_tiles, true)
```
Duplicate line.

grilledham commented on map_gen/maps/HilbertSandTrap.lua:
```diff
@@ -0,0 +1,287 @@
+			player.surface.set_tiles(replace_tiles, true)
+		end
+		player.surface.set_tiles(replace_tiles, true)
+		player.insert {name = event.item.name, count = refund_count}
```
`player.insert {name = event.item.name, count = refund_count}` should be inside the if block above. It was a mistake in the hotfix.

mheguy commented on map_gen/maps/HilbertSandTrap.lua:
```diff
@@ -0,0 +1,287 @@
+		MGSP.grass_only,
+		ore_settings,
+		biter_settings,
+		starting_area,
```
Since you already have the presets loaded you may as well use them and save lines. :)
```suggestion
		MGSP.starting_area_very_low,
```

mheguy commented on map_gen/maps/HilbertSandTrap.lua:
```diff
@@ -0,0 +1,287 @@
+        MGSP.cliff_none,
+		MGSP.grass_only,
+		ore_settings,
+		biter_settings,
```
```suggestion
		MGSP.enemy_very_high,
```

SimonFlapse commented:
Given the first play through, I'd say we need to override hydra settings for this map, before merging it

R-Nukem committed:
Update HilbertSandTrap.lua

Updated to include suggestions and hotfixes and generally clean up the code a little.

R-Nukem commented:
Code has been cleaned up a bit. Still working on converting to Plague's entity restriction module and will update again when that is finished and tested.

R-Nukem committed:
Update HilbertSandTrap.lua

R-Nukem committed:
Update HilbertSandTrap.lua

Combined ghost and tile-ghost handling

mheguy commented on map_gen/maps/HilbertSandTrap.lua:
```diff
@@ -0,0 +1,282 @@
+RS.set_map_gen_settings(
+    {
+        MGSP.cliff_none,
+		MGSP.grass_only,
```
Because of changes in 17, you will need to add another setting when using `grass_only`: `enable_water`

R-Nukem replied:
Thanks for the heads-up. This has been changed in the most recent commit.

R-Nukem replied:
Thanks for the heads-up. This has been changed in the most recent commit.

R-Nukem committed:
Update HilbertSandTrap.lua

added enable_water mapgen setting for use with grass_only
